packages/tuono-router/src/components/RouterContext.tsx

```diff
@@ -20,3 +36,88 @@ export function getRouterContext(): React.Context<Router> {
 
   return routerContext
 }
+
+interface RouterContextProviderProps {
+  router: Router
+  children: ReactNode
+  serverSideProps?: ServerRouterInfo
+}
+
+function getInitialLocation(
+  serverSideProps?: ServerRouterInfo,
+): ParsedLocation {
+  if (typeof document === 'undefined') {
+    return {
+      pathname: serverSideProps?.pathname || '',
+      hash: '',
+      href: serverSideProps?.href || '',
+      searchStr: serverSideProps?.searchStr || '',
+      // TODO: Polyfill URLSearchParams
+      search: {},
+    }
+  }
+
+  const { location } = window
+  return {
+    pathname: location.pathname,
+    hash: location.hash,
+    href: location.href,
+    searchStr: location.search,
+    search: Object.fromEntries(new URLSearchParams(location.search)),
+  }
+}
+
+export function RouterContextProvider({
+  router,
+  children,
+  serverSideProps,
+}: RouterContextProviderProps): ReactNode {
+  // Allow the router to update options on the router instance
+  router.update({ ...router.options } as Parameters<typeof router.update>[0])
+
+  const [location, updateLocation] = useState<ParsedLocation>(
```

Comment:
```suggestion
  const [location, setLocation] = useState<ParsedLocation>(
```

This can be renamed when creating the `RouterContext` value to useLocation, but let's stick to react conventions when declaring state variables